### PR TITLE
Added support for Animeworld.tv

### PR DESCRIPTION
**Pull request by n4y0n**
Just tought of adding support for the italian anime streaming site Animeworld.tv

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -292,6 +292,10 @@ Makes it possible to use your MyAnimeList/Anilist/Kitsu/Simkl anime/mangalist as
                 <td><a href="https://www.animetoast.cc"><img src="https://favicon.malsync.moe/?domain=https://www.animetoast.cc"> Animetoast</a></td>
```
Comment (henrik9999) [inline README.md:292]: this file is autogenerated on release please dont edit

File: pages.md [hunk 2]
```diff
@@ -166,6 +166,14 @@
     <td>:x:</td>
     <td>:x:</td>
     <td>:x:</td>
+  </tr><tr>
```
Comment (henrik9999) [inline pages.md:169]: this file is autogenerated on release please dont edit

File: src/pages/list.json [hunk 3]
```diff
@@ -673,5 +673,10 @@
     "domain": "https://luminousscans.com",
     "type": "manga",
```
Comment (henrik9999) [inline src/pages/list.json:674]: this file is autogenerated on release please dont edit

**Comment by lolamtisch** [comment]
Please dont edit README.md pages.md list.json, because they are autogenerated at release.

File: src/pages/Animeworld/main.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          .$('div.widget-body>div.server.active>ul.episodes.range.active>li.episode>a.active')
+          .text()
+          .trim(),
+      );
+      if (episode) return episode;
+      return 1;
+    },
+    uiSelector(selector) {
+      j.$('span.clearfix').first().after(j.html(selector));
+    },
```
Comment (henrik9999) [inline src/pages/Animeworld/main.ts:36]: the site has anilist and myanimelist urls on the page so please implement the function getMalUrl

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/Animeworld/main.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    getIdentifier(url) {
+      return utils.urlPart(url, 4);
+    },
+    getOverviewUrl(url) {
+      return `${Animeworld.domain}/${Animeworld.sync.getIdentifier(url)}`;
+    },
+    getEpisode(url) {
+      const episode = Number(
+        j
+          .$('div.widget-body>div.server.active>ul.episodes.range.active>li.episode>a.active')
```
Comment (lolamtisch) [inline src/pages/Animeworld/main.ts:27]: This selector will break with every small page html change

Comment (n4y0n) [reply src/pages/Animeworld/main.ts:27]: I'll change it 👍

File: README.md (hunk 1 above)
Comment (n4y0n) [reply README.md]: I reverted the file back to master and did the same for the other autogenerated ones.

File: src/pages/Animeworld/main.ts (hunk 5 above)
Comment (n4y0n) [reply src/pages/Animeworld/main.ts]: I hope now it could work better

File: src/pages/Animeworld/main.ts [hunk 6]
```diff
@@ -21,6 +21,17 @@ export const Animeworld: pageInterface = {
     getOverviewUrl(url) {
       return `${Animeworld.domain}/${Animeworld.sync.getIdentifier(url)}`;
     },
+    getMalUrl(provider) {
+      return new Promise(resolve => {
+        if (provider === 'ANILIST') {
+          resolve(j.$('#mal-button').attr('href') || false);
+        }
+        if (provider === 'MAL') {
```
Comment (henrik9999) [inline src/pages/Animeworld/main.ts:29]: it should always return mal if its available like on the other implementations

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/Animeworld/main.ts (hunk 6 above)
Comment (n4y0n) [reply src/pages/Animeworld/main.ts:29]: Understood 👍

File: src/pages/Animeworld/main.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    getIdentifier(url) {
+      return utils.urlPart(url, 4);
+    },
+    getOverviewUrl(url) {
+      return `${Animeworld.domain}/${Animeworld.sync.getIdentifier(url)}`;
+    },
+    getMalUrl(provider) {
+      return new Promise(resolve => {
+        resolve(j.$('#mal-button').attr('href') || false);
```
Comment (henrik9999) [inline src/pages/Animeworld/main.ts:26]: i think you misunderstood what im trying to say
https://github.com/MALSync/MALSync/blob/master/src/pages/animepahe/main.ts#L52
i mean it like this
so mal first anilist if you use anilist second

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/Animeworld/main.ts (hunk 7 above)
Comment (n4y0n) [reply src/pages/Animeworld/main.ts:26]: ahh got it, sorry i misunderstood

**Comment by n4y0n** [comment]
Now it should look like this
image

I don't think i'll add anything else.

**Review by n4y0n — COMMENTED** [review]
So... yeah 😊

**Comment by henrik9999** [comment]
dont do anything i will fix it myself

**Review by henrik9999 — APPROVED** [review]

**Pull request merged**